### DISCUSSION TITLE
v8.0.0 installation problem

### DISCUSSION THREAD

**Discussion by ardao13** [su2code/SU2#2128] [General]
How can I solve the following problem?

![Screenshot from 2023-09-08 14-54-24](https://github.com/su2code/SU2/assets/73038236/045ca49b-f093-49e0-8573-ea487ad64753)

[](url)

**Comment by jblueh** [comment]
Hello @mardar572, could you please provide additional details? How did you configure SU2 with meson? What is the output of meson? Which compiler do you use?

**Reply by ardao13** [reply]
hi @jblueh,

I am using openmpi407. There is no problem of installation of SU2  for version 751 and 740.

**meson input:**
meson.py build -Dwith-mpi=enabled -Denable-autodiff=true -Denable-directdiff=true  --prefix=.../apps/su2

[meson_out.txt](https://github.com/su2code/SU2/files/12559713/meson_out.txt)

**Reply by jblueh** [reply]
The meson output shows that you are using a very old version of GCC (4.8.5), this might be the issue. Do you have the option to try a more recent version of GCC?

**Reply by ardao13** [reply]
I have installed GCC (12.2) and try with it but it didn't solve the issue. Then I add the **-Denable-tecio=false** option to meson input. It helped the passed previous step but now I am having the following error :(

As stated earlier I can successfully install SU2 751 without this option  **-Denable-tecio=false**

![Annotation 2023-09-08 161032](https://github.com/su2code/SU2/assets/73038236/6ab39ce0-b7d7-4fd7-ad5d-3c2ae664411f)

**Reply by jblueh** [reply]
Could you double-check the meson output? Does meson actually use your GCC 12.2?

**Reply by ardao13** [reply]
@jblueh thank you very much for your lead. 

The problem solved when I set **CC** env variable to /path_of_mpi/bin/mpicc

**Reply by jblueh** [reply]
Yes, this can happen if there are multiple GCCs on the system, setting `CC` and/or `CXX` to point to the right one is the way to go. Glad to hear it works now!

**Comment by jaspe55** [comment]
I just tried to install the latest SU2 8.00 release to my HPC account, and
when building it using meson.py I got this message:
...
ninja executable not found. Building ...
bootstrapping ninja...
wrote build.ninja.
bootstrap complete.  rebuilding...
[31/31] LINK ninja
Traceback (most recent call last):
  File
"/scratch/677131/S800/SOURCE/SU2/externals/meson/mesonbuild/mesonmain.py",
line 146, in run
    return options.run_func(options)
  File
"/scratch/677131/S800/SOURCE/SU2/externals/meson/mesonbuild/msetup.py",
line 294, in run
    app.generate()
  File
"/scratch/677131/S800/SOURCE/SU2/externals/meson/mesonbuild/msetup.py",
line 184, in generate
    with mesonlib.BuildDirLock(self.build_dir):
  File
"/scratch/677131/S800/SOURCE/SU2/externals/meson/mesonbuild/mesonlib/posix.py",
line 32, in __enter__
    fcntl.flock(self.lockfile, fcntl.LOCK_EX | fcntl.LOCK_NB)
OSError: [Errno 38] Function not implemented

ERROR: Unhandled python exception

    This is a Meson bug and should be reported!

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 I have tested some python3 versions :  3.6, 3.97, 3.7.16.
For 3.6 version, it is old and continuation is rejected: Ok!
For the versions > 3.7, the message is the same shown above!

   Kind regards,

        Jairo

**Reply by bigfooted** [reply]
Did this work for you with older versions? The final flock error could mean that on your cluster, some file locking functionality has been disabled. I see that you are compiling in /scratch. Do you have another filesystem where you can compile? For instance inside /home instead of /scratch?

**Comment by jaspe55** [comment]
Hello Nijso, Unfortunately, not so far: My home directory is
/scratch/677131. I will talk to HPC sysadmin to see whether there is or not
another directory available, at least for a test! /scratch is based on
cephfs/NFS.

**Comment by jaspe55** [comment]
Supplementary info: Yes, for SU2 7.31 I was able to install it on another
cluster , in 2022.

**Comment by jaspe55** [comment]
News: I just tried install SU2V801 in another cluster, and no problems with
locking files on it!
Ninja has been aborted on the compilation:

...

                 from
../SU2_CFD/src/output/filewriter/CParaviewFileWriter.cpp:28:
../externals/codi/include/codi/tapes/data/chunk.hpp:107:32: warning: inline
function ‘virtual void codi::ChunkBase::deleteData()’ used but never defined
       CODI_INLINE virtual void deleteData() = 0;              ///< Delete
the allocated data.
                                ^
../externals/codi/include/codi/tapes/data/chunk.hpp:106:32: warning: inline
function ‘virtual void codi::ChunkBase::allocateData()’ used but never
defined
       CODI_INLINE virtual void allocateData() = 0;            ///<
Allocated the data if it was deallocated before.
                                ^
[538/1371] Compiling C++ object Common/src/libSU2Common.a.p/CConfig.cpp.o
ninja: build stopped: subcommand failed.

Info on compiler:
***@***.*** SU2]$ echo $CC
/opt/ohpc/pub/mpi/openmpi-gnu/1.10.7/bin/mpicc
***@***.*** SU2]$ echo $CXX
/opt/ohpc/pub/mpi/openmpi-gnu/1.10.7/bin/mpicxx
***@***.*** SU2]$ mpicc -v
Using built-in specs.
COLLECT_GCC=/opt/ohpc/pub/compiler/gcc/5.4.0/bin/gcc
COLLECT_LTO_WRAPPER=/opt/ohpc/pub/compiler/gcc/5.4.0/libexec/gcc/x86_64-unknown-linux-gnu/5.4.0/lto-wrapper
Target: x86_64-unknown-linux-gnu
Configured with: ../configure --disable-multilib
--enable-languages=c,c++,fortran --prefix=/opt/ohpc/pub/compiler/gcc/5.4.0
Thread model: posix
gcc version 5.4.0 (GCC)
***@***.*** SU2]$ mpicxx -v
Using built-in specs.
COLLECT_GCC=/opt/ohpc/pub/compiler/gcc/5.4.0/bin/g++
COLLECT_LTO_WRAPPER=/opt/ohpc/pub/compiler/gcc/5.4.0/libexec/gcc/x86_64-unknown-linux-gnu/5.4.0/lto-wrapper
Target: x86_64-unknown-linux-gnu
Configured with: ../configure --disable-multilib
--enable-languages=c,c++,fortran --prefix=/opt/ohpc/pub/compiler/gcc/5.4.0
Thread model: posix

**Comment by jaspe55** [comment]
FYI installation worked when I updated to GCC 8.8.3, but when I star the
flow solver, I got the following error: SU2_CFD: error while loading shared
libraries: libmpi.so.12: cannot open shared object file: No such file or
directory!
